   # CAPSTONE PROJECT 

#### PROBLEM Statement :
- Critic overview on the Human Development Index on Environmental Safety and Economic health

Theory:
The United Nations Development Program have set an index called the Human Development Index ‘HDI’ to masseur the development and life quality in each country on detailed attributes within three main aspects of human life: Health, Education and Income.
The importance within this index relies on the fact that it sets the role models countries for other countries to follow it polices and guidelines in order to active the same level of development as well, so the index should be reliable and well generlized.

My theory states that NOT every country that increases in it HDI will contribute positively with Environmental, personal safety and Economic health in which will make the HDI is questionable and not sufficient to make a fair or reliable judgment for the development within these countries.
Espicaly the Arabic countries in the environmental part.

**Approach:**

- **First critic overview, Environmental**
- I will take the countries with HDI increase rate between 2012 and 2017,compare it to the PM2.5 air pollution exposure.
- Exposure to PM2.5 has multiple short term and long term health impacts, A prolonged exposure to PM2.5 can cause permanent respiratory problems such as asthma, chronic bronchitis and heart disease.
Research shows that every 10ug/m3 increase in PM 2.5, increases all-cause mortality between 3-26%, chances of childhood asthma by 16%, chances of lung cancer by 36% and heart attacks by 44%.

- **Second critic overview, Personal Safety index**
- The countries that passed the first critic overview will be taken to see the rates of the personal safety percentage change within the five years.
-The personal safety index is a percentage that Mauser the scores of Homicide rate, Perceived criminality, Political killings and torture, Traffic deaths.
This index is developed by The Social Progress Imperative - Washington, DC.

- **Third critic overview,, -Economic health:**

- The countries that passed the first and second critic overview will be uptaken to see state of it debt-to-GDP percentage.

- The debt-to-GDP percentage is the metric comparing a country's public debt to its gross domestic product (GDP). By comparing what a country owes with what it produces, the debt-to-GDP ratio reliably indicates that particular country’s ability to pay back its debts.

- i will see which countries that manged to reduce the dept and then i will fetch the healthier economic one by taking the countries with less than 77% debt-to-GDP, as a study by the World Bank found that if the debt-to-GDP ratio of a country exceeds 77% for an extended period of time, it slows economic growth and puts the countries in higher economic risks.

- **Intervention:**
- intervention will be discussed with the conclusion

# Importting libraries 

In [1453]:
import pandas as pd
from matplotlib import pyplot as plt
import seaborn as sns
import numpy as np
import warnings
warnings.filterwarnings('ignore')
from scipy import stats
sns.set_style('white')
co=['#174147','#98B6C1','#2C8592','#BFD1D7','#05BCD8','#96E4F0','#216772','#B2E0F0']
import patsy
import matplotlib.pyplot as plt
plt.style.use('fivethirtyeight')
sns.set(font_scale=1.5)
%matplotlib inline
%config InlineBackend.figure_format = 'retina'

# Part one - HDI

## Importing the human developmnt index -HDI- 

In [1474]:
hdi= pd.read_csv('HDI_.csv').round(2)

In [1475]:
hdi = hdi.set_index(['Country_'])

### Find the countries that incresed in the HDI rate in the on 2012-2017

In [1476]:
hdi['change_rate'] =  (hdi.HDI_2017-hdi.HDI_2012)/hdi.HDI_2012*100

### Take the countries with incresed HDI

In [1477]:
subset_high_C = hdi[(hdi['change_rate'] >= 0)] #subset

In [1478]:
subset_high_C.shape #check shape

(156, 10)

In [1459]:
subset_high_C.round(2).head() #check the datafram

HDI_2010  HDI_2011  HDI_2012  HDI_2013  HDI_2014  HDI_2015  \
Country_                                                                  
Afghanistan      0.46      0.47      0.48      0.49      0.49      0.49   
Albania          0.74      0.75      0.77      0.77      0.77      0.78   
Algeria          0.73      0.74      0.74      0.74      0.75      0.75   
Argentina        0.81      0.82      0.82      0.82      0.82      0.82   
Armenia          0.73      0.73      0.74      0.74      0.74      0.75   

             HDI_2016  HDI_2017      Country  change_rate  
Country_                                                   
Afghanistan      0.49      0.50  Afghanistan         4.17  
Albania          0.78      0.78      Albania         1.30  
Algeria          0.75      0.75      Algeria         1.35  
Argentina        0.82      0.82    Argentina         0.00  
Armenia          0.75      0.76      Armenia         2.70

#### Out of 160 countries, 156 countries incresed in thier HDI , 

Now i will just check the countries with decresed HDI

In [1359]:
subset_low_c = hdi[(hdi['change_rate'] < 0)]

In [1360]:
subset_low_c

HDI_2010  HDI_2011  HDI_2012  HDI_2013  HDI_2014  HDI_2015  \
Country_                                                                
Libya          0.76      0.71      0.74      0.71      0.70      0.69   
Syria          0.64      0.64      0.63      0.57      0.55      0.54   
Venezuela      0.76      0.77      0.77      0.78      0.78      0.78   
Yemen          0.50      0.50      0.50      0.51      0.50      0.48   

           HDI_2016  HDI_2017    Country  change_rate  
Country_                                               
Libya          0.69      0.71      Libya    -4.054054  
Syria          0.54      0.54      Syria   -14.285714  
Venezuela      0.77      0.76  Venezuela    -1.298701  
Yemen          0.46      0.45      Yemen   -10.000000

it is logical for these countries to have lower HDI regarding thier war and political distress.

## Now i know the 156 countries that incresed in the HDI, lets look up their PM2.5 rates  

In [1470]:
#taking the incresed HDI countries for the next stage and look up the pm2.5 data for them.
pm_countries = pd.DataFrame()
pm_countries['countries'] = subset_high_C.index
#pm_countries.to_csv('pm_countries.csv')

_________________________

______________

# Part Two - PM2.5 air polution 
i got the data for the countries that passed the first critic overview, however,<font color="red">,there was 20 Countries with NO PM2.5 data </font>
- PM2.5 air pollution fetched is masuerd by the mean annual exposure (micrograms per cubic meter)

- what i am going to check ? 
 - First, does all incresed  HDI countries contrubuited to the envairoment positivly ?
 - second, how many ararbic countries contruibitd positivly ?

   lets see ! 

###  Import the P2.5 data 

In [1512]:
air = pd.read_csv('pm2.5_.csv').round(3)

In [1513]:
air.shape

(136, 11)

In [1514]:
air = air.set_index(['Country_'])

### Define the arabic countries 

In [1515]:
arabic_countries=['Jordan','United Arab Emirates','Tunisia','Sudan','Kuwait','South Sudan','Iraq','Saudi Arabia','Qatar','Oman','Morocco','Mauritania','Egypt, Arab Rep.','Bahrain','Algeria']

In [1516]:
test =[]
for i in air.index:
    if i in arabic_countries:
        test.append(1)
    else:
        test.append(0)

In [1517]:
air['arabic'] = test

In [1518]:
arabic = air[air['arabic']==1]

###  Massure the change rate betweent 2012-2017

In [1519]:
air['change_rate'] =  (air.pm_2017-air.pm_2012)/air.pm_2012 *100

In [1520]:
air['diffreance'] =  (air.pm_2017-air.pm_2012)

####  Distinguish between the incresed pm2.5 and the decrese pm2.5 

In [1521]:
incresed_p = air[(air['change_rate'] >= 0)]

In [1522]:
incresed_p.index.value_counts().sum() # countries with incresed pm2.5 rate

45

In [1523]:
arabic_incresed = incresed_p[incresed_p['arabic']==1]

In [1524]:
decresed_P = air[(air['change_rate'] < 0)].round(2) 

In [1525]:
decresed_P.index.value_counts().sum()# countries with decresed pm2.5 rate

91

In [1526]:
# Idenitify highest decrese in pm2.5 pollution 

In [1527]:
decresed_P.sort_values(by="diffreance" ).head(5) # top 10 in the highest decrese.

code  pm_2010  pm_2011  pm_2012  pm_2013  pm_2014  pm_2015  \
Country_                                                                 
Sri Lanka    LKA    31.03    31.23    30.08    28.90    26.92    25.34   
China        CHN    69.48    70.54    63.83    65.52    59.77    59.06   
Bangladesh   BGD    70.76    70.21    70.79    65.64    68.40    66.97   
Afghanistan  AFG    65.25    66.84    66.02    61.37    59.01    61.25   
Vietnam      VNM    39.36    38.49    37.23    36.84    34.90    32.47   

             pm_2016  pm_2017      Country  arabic  change_rate  diffreance  
Country_                                                                     
Sri Lanka      14.02    11.10    Sri Lanka       0       -63.10      -18.98  
China          52.21    52.66        China       0       -17.49      -11.16  
Bangladesh     60.13    60.85   Bangladesh       0       -14.05       -9.95  
Afghanistan    56.29    56.91  Afghanistan       0       -13.80       -9.11  
Vietnam        30.24    29.63      Vietnam       0       -20.42       -7.60

In [1528]:
# outside reserch to do - find out what those countries were doing to get this decreese in the pm2.5

In [1529]:
arabic_decresed = decresed_P[decresed_P['arabic']==1]

In [1530]:
arabic_decresed 

code  pm_2010  pm_2011  pm_2012  pm_2013  pm_2014  pm_2015  pm_2016  \
Country_                                                                       
Iraq      IRQ    68.71    66.36    66.84    63.80    56.57    66.26    63.96   
Kuwait    KWT    62.74    62.82    64.07    62.61    55.62    65.41    60.84   

          pm_2017 Country  arabic  change_rate  diffreance  
Country_                                                    
Iraq        61.64    Iraq       1        -7.79       -5.21  
Kuwait      60.74  Kuwait       1        -5.19       -3.33

### lets arrange the first theory findings in a data frame ! 

In [1531]:
F_Theory= pd.DataFrame()

In [1532]:
F_Theory['Incresed%'] = [incresed_p.index.value_counts().sum()/air.index.value_counts().sum() *100]

In [1533]:
F_Theory['Decresed%'] = [decresed_P.index.value_counts().sum()/air.index.value_counts().sum()*100]

In [1534]:
F_Theory['Incresed_arabic']= arabic_incresed.index.value_counts().sum()
F_Theory['Decresed_arabic'] = arabic_decresed.index.value_counts().sum()

In [1535]:
F_Theory

Incresed%  Decresed%  Incresed_arabic  Decresed_arabic
0  33.088235  66.911765               13                2

First Theory was proven true : 
    - Not all countries contrubuit postivily with the HDI incresemnt 
    - 13!! out of 15 arabic countries contuorbted negetivly !

 91 countries only contirubated +ve-ly towards the enviroment, of which only 2 were arabic. 
### Now lets take the 91 countries with postive contribution towards envaromental health for the next theory :
- personal saftey

In [1536]:
ps_countries = pd.DataFrame()
ps_countries['countries'] = decresed_P.index
#pm_countries.to_csv('ps_countriess.csv')

   --------------------

----------

# Third Part : Personal Saftey Index

Personal saftey index is a percentage that masuer a calculated scores of Homicide rate,Perceived criminality,Political killings and torture,Traffic deaths, how ever, there was 3 countries with no PSI data

- what i am going to check ? 
 - First, does all incresed  HDI countries contrubuited to the personal saftey index positivly ?

   lets see ! 

In [1590]:
ps = pd.read_csv('personal_saftey.csv')

In [1591]:
ps.set_index('Country').head(1)

Code  ps_2013  ps_2014  ps_2015  ps_2016  ps_2017  ps_2018
Country                                                               
Afghanistan  AFG     45.6    43.88     44.4    44.23    45.63    46.85

In [1592]:
ps.shape

(89, 8)

In [1593]:
ps['Personl_change_rate']= (ps['ps_2018']- ps['ps_2013'])/ps['ps_2013']*100

In [1594]:
decrese_ps = ps[(ps['Personl_change_rate'] <0)] # countries with decresed peronal saftey
decrese_ps.shape

(35, 9)

In [1595]:
incresed_ps = ps[(ps['Personl_change_rate'] >0)] # countries with decresed peronal saftey
incresed_ps.shape

(54, 9)

In [1596]:
incresed_ps.sort_values(by="Personl_change_rate",ascending=False).head(1) # top 10 in the +ve change rate 

Country Code  ps_2013  ps_2014  ps_2015  ps_2016  ps_2017  ps_2018  \
76  Sri Lanka  LKA    64.67    65.86    69.32    71.79    72.32    73.28   

    Personl_change_rate  
76            13.313747

In [1597]:
decrese_ps.sort_values(by="Personl_change_rate").head(1)

Country Code  ps_2013  ps_2014  ps_2015  ps_2016  ps_2017  ps_2018  \
55  Montenegro  MNE    74.64    74.78    71.66    69.29    69.44    66.14   

    Personl_change_rate  
55           -11.387996

Lets arrange the findings in a dataframe !

In [1598]:
Theory_2 = pd.DataFrame()

In [1599]:
Theory_2['Incresed%'] = [incresed_ps.index.value_counts().sum()/ps.index.value_counts().sum() *100]

In [1600]:
Theory_2['Decresed%'] = [decrese_ps.index.value_counts().sum()/ps.index.value_counts().sum() *100]

In [1601]:
Theory_2

Incresed%  Decresed%
0  60.674157  39.325843

### Now lets take the 54 countries that now we know they passed both the first and second critic overviews to the last part.



In [1612]:
#third_t = incresed_ps.to_csv('third_t__.csv')

------------

----------

# Fourth part- Debt-to-GDP


The debt-to-GDP percentage is the metric comparing a country's public debt to its gross domestic product (GDP). By comparing what a country owes with what it produces, the debt-to-GDP ratio reliably indicates that particular country’s ability to pay back its debts, however,<font color="red"> there was one Country with no dept to gpd data that represent </font>

- what i am going to check ? 
 - First, does all countries that now have passed the pm.2.5 and personal saftey and had a incresed hdi will contrubuited positivly to the debt-to-GDP percentage ?
 - second, take healthier countries that had less than 77% of ebt-to-GDP percentage to be the finale winners in this project ! 

   lets see ! 

In [1659]:
dept = pd.read_csv('dept_to_gpd.csv')

In [1660]:
dept.shape

(53, 8)

In [1635]:
dept.columns

Index(['Country', 'dt_2012', 'dt_2013', 'dt_2014', 'dt_2015', 'dt_2016',
       'dt_2017', 'dt_2018'],
      dtype='object')

In [1636]:
dept['change_rate_dt']= (dept['dt_2017']- dept['dt_2012'])/dept['dt_2012']*100

In [1637]:
decrese_dept = dept[(dept['change_rate_dt'] <0)] # countries with decresed dept
decrese_dept.shape

(16, 9)

In [1638]:
increse_dept = dept[(dept['change_rate_dt'] >0)] # countries with decresed dept
increse_dept.shape

(37, 9)

In [1645]:
Theory_3 = pd.DataFrame()

In [1647]:
Theory_3['Incresed%'] = [increse_dept.index.value_counts().sum()/dept.index.value_counts().sum() *100]
Theory_3['Decresed%'] = [decrese_dept.index.value_counts().sum()/dept.index.value_counts().sum() *100]

In [1649]:
Theory_3

Incresed%  Decresed%
0  69.811321  30.188679

70% of these countries actully incresed in thier dept......

_________

In [1653]:
#fined the healthieist countries 

In [1654]:
#Remove countries higher than 77% dept to gpd 

In [1655]:
healthy_countries = decrese_dept[decrese_dept['dt_2018'] <= 77]

In [1656]:
healthy_countries.shape

(15, 9)

In [1657]:
healthy_countries.shape

(15, 9)

In [1658]:
healthy_countries.sort_values(by='change_rate_dt').head()

Country  dt_2012  dt_2013  dt_2014  dt_2015  dt_2016  dt_2017  \
18         Iceland   89.369   81.789   79.727   65.955   51.697   39.958   
21         Ireland  119.925  119.817  104.265   76.931   73.556   68.555   
10  Czech Republic   44.470   44.909   42.169   39.955   36.806   34.669   
11         Denmark   44.894   44.046   44.270   39.926   37.868   35.294   
33     Netherlands   65.511   67.005   67.100   63.970   61.299   56.450   

    dt_2018  change_rate_dt  
18   37.022      -55.288747  
21   66.577      -42.835105  
10   33.166      -22.039577  
11   34.739      -21.383704  
33   53.099      -13.831265

# These are the countries which succesed in :
   ####  *Maintain increasing their HDI.
   #### * Maintain decresing their pm.2.5 rates.
   #### * Maintain increasing their personal saftey index.
   #### * Maintain  decresing their dept and was less than 77%.
    
 ### which only represent only 9.37% of the countries with incresed HDI  

# Remaining things to do : 
    - Visualztion of the data
    - highlihgt hightest countries for postiv changes and look up thier intrvention for how did they make it.